src/erl_optics.erl

```diff
@@ -315,8 +322,42 @@ create_optics(Name) ->
     end.
 
 get_lens(Key) ->
-    foil:lookup(?NS, Key).
-
+    case foil:lookup(?NS, Key) of
+        {ok, Ptr} ->
+            {ok, Ptr};
+        Err ->
+            {error, {Err,Key}}
```

Comment:
It looks like this would make error tuples that look like `{error, {{error, module_not_found | foil_not_started}, Key}`. I think it'd be cleaner to match on the errors you can receive from `foil:lookup`. https://github.com/lpgauth/foil/blob/master/src/foil.erl#L88